src/oscar/apps/dashboard/offers/forms.py

```diff
@@ -188,4 +203,30 @@ def save(self, *args, **kwargs):
 
 class OfferSearchForm(forms.Form):
     name = forms.CharField(required=False, label=_("Offer name"))
-    is_active = forms.BooleanField(required=False, label=_("Is active?"))
+    is_active = forms.NullBooleanField(required=False, label=_("Is active?"))
+    offer_type = forms.ChoiceField(required=False, label=_("Offer type"), choices=())
+    has_vouchers = forms.NullBooleanField(required=False, label=_("Has vouchers?"))
+    voucher_code = forms.CharField(required=False, label=_("Voucher code"))
+
+    basic_fields = [
+        'name',
+        'is_active',
+    ]
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.fields['offer_type'].choices = ((("", "---------"),)
+                                             + tuple(choice for choice in ConditionalOffer.TYPE_CHOICES
+                                                     if choice[0] in get_implemented_offer_type_values()))
+        # Give the "unknown" choice a more meaningful label than the default of "Unknown"
+        NULL_BOOLEAN_FIELD_CHOICES = (
+            ('unknown', _('---------')),
+            ('true', _('Yes')),
+            ('false', _('No')),
+        )
+        self.fields['is_active'].widget.choices = NULL_BOOLEAN_FIELD_CHOICES
+        self.fields['has_vouchers'].widget.choices = NULL_BOOLEAN_FIELD_CHOICES
```

Comment:
Is it really necessary to do this in the constructor? Can it not be specified in the form widget definition?